runtime/sqlite_record_logger.hpp

```diff
@@ -180,7 +180,7 @@ class sqlite_record_logger : public record_logger {
 		if (result != registered_tables.cend()) {
 			return result->second;
 		} else {
-			const std::lock_guard lock{_m_registry_lock};
+			const std::shared_lock lock{_m_registry_lock};
```

Comment:
This method needs to look like this:

```C++
	sqlite_thread& get_sqlite_thread(const record& r) {
		const record_header& rh = r.get_record_header();
		{
			const std::shared_lock{_m_registry_lock};
			auto result = registered_tables.find(rh.get_id());
			if (result != registered_tables.cend()) {
				return result->second;
			}
		}
		const std::unique_lock lock{_m_registry_lock};
		auto pair = registered_tables.try_emplace(rh.get_id(), rh);
		return pair.first->second;
	}
```

I am guarding `registered_tables.find` and its returned value with `shared_lock`, since it is a read.

I am guarding `registered_tables.try_emplace` with `unique_lock`, since that is a write.

Reply:
Is it similar for all the methods using shared_lock, i mean the localization of the process? 
Cause the previous code used to be this:

```cpp
sqlite_thread& get_sqlite_thread(const record& r) {
		const record_header& rh = r.get_record_header();
		auto result = registered_tables.find(rh.get_id());
		if (result != registered_tables.cend()) {
			return result->second;
		} else {
			const std::shared_lock lock{_m_registry_lock};
			auto pair = registered_tables.try_emplace(rh.get_id(), rh);
			return pair.first->second;
		}
	}
```
And the change you've asked is:
```cpp
sqlite_thread& get_sqlite_thread(const record& r) {
		const record_header& rh = r.get_record_header();
		{
			const std::shared_lock{_m_registry_lock};
			auto result = registered_tables.find(rh.get_id());
			if (result != registered_tables.cend()) {
				return result->second;
			}
		}
		const std::unique_lock lock{_m_registry_lock};
		auto pair = registered_tables.try_emplace(rh.get_id(), rh);
		return pair.first->second;
	}
```
I am assuming it is, and I'm changing the locks in that format if otherwise, I'll change it back.

Reply:
The other instances of `shared_lock` I think already distinguish the reads from the writes. It's just in this place (as far as I know) that they were mixed together.